akka-testkit/src/main/scala/akka/testkit/TestKit.scala

```diff
@@ -292,19 +297,23 @@ trait TestKitBase {
    * Note that the timeout is scaled using Duration.dilated,
    * which uses the configuration entry "akka.test.timefactor".
    */
-  def awaitAssert(a: ⇒ Any, max: Duration = Duration.Undefined, interval: Duration = 100.millis) {
+  def awaitAssert[A](a: ⇒ A, max: Duration = Duration.Undefined, interval: Duration = 100.millis): A = {
```

Comment:
should be source compatible, and we have the disclaimer about testkit for bincompat

Reply:
Exactly my thinking, and tremendously useful :)

Reply:
Nice, high five 👍